frontend/src/utils/game_utils.js

```diff
@@ -178,6 +178,15 @@ export const getAdjacentTilePath = (tower, end) => {
     return Array(SAW_TICK_DURATION).fill([avgX, avgY, tower.z + (tower.height/2)]);
 }
 
+export const getUpwardPath = (tower, speed = 0.04, steps = 75) => {
+    const [x, y, z] = tower.position;
+    const path = [[x, y, z + tower.height]];
+    for (let i = 0; i < steps; i++) {
+        path.push([x, y, z + tower.height + (i * speed)]);
+    }
```

Comment:
This suggestion avoids the multiplication. Also, it wasn't clear if it was your intention to double the initial position (i starts at 0) but that could be fixed if needed by starting `path` as an empty list or reordering the increment.

```suggestion
    let height = z + tower.height;
    for (let i = 0; i < steps; i++) {
        path.push([x, y, height]);
        height += speed
    }
```